### ISSUE TITLE
IllegalStateException

### ISSUE THREAD

**Issue by Sylux6** [Kodehawa/imageboard-api#60]
Using this code throws an exception

```kotlin
DefaultImageBoards.DANBOORU.search()
```
Error message: `User-Agent must be initialized to make requests, Initialize with ImageBoard.setUserAgent()`

**Comment by Kodehawa** [comment]
User-Agent must be initialized to make requests, Initialize with ImageBoard.setUserAgent()

**Comment by Sylux6** [comment]
Yes I know but shouldn't the API wrapper use a working User-Agent by default? 
It just sounds kinda odd that is not usable out of the box. What do you think?

**Comment by Sylux6** [comment]
Maybe this should be added to the README as it's not usable out of the box because of the default agent being blocked

**Comment by Kodehawa** [comment]
Added so. 

To your previous comment (which you deleted, I think?), setting a default UA just leads to being UA blocked. We did it before. I do suspect it's because some people spammed imageboards with this API wrapper, which isn't something I can prevent against, but it does set some flags in, say, cloudflare.

Better to set your own!

**Comment by Sylux6** [comment]
Yes, exactly what I thought after writing my comment we can't avoid this I guess :+1:
